provider/disk_utilities.py

```diff
@@ -0,0 +1,57 @@
+"""
+Disk utilities functions for managing disks and filesystems in guest environments.
+
+This module contains functions related to disk detection, filesystem creation,
+and command execution for various OS types.
+"""
+
+from virttest import error_context, utils_disk, utils_misc
+from virttest.utils_misc import get_linux_drive_path
+from virttest.utils_windows.drive import get_disk_props_by_serial_number
+
+
+def get_window_disk_index_by_serial(serial, session):
+    idx_info = get_disk_props_by_serial_number(session, serial, ["Index"])
+    if idx_info:
+        return idx_info["Index"]
+
```

Comment:
I suggest adding a function like get_disk_by_id or get_disk_by_serial. The ID may be the serial/www/device ID. It may handle different OSs.  for windows it return disk idx, for linux , it return disk name

Reply:
@qingwangrh done!